dashboard/test/lib/lti_access_token_test.rb

```diff
@@ -0,0 +1,43 @@
+require 'test_helper'
+require 'date'
+
+class LtiAccessTokenTest < ActiveSupport::TestCase
+  include LtiAccessToken
+
+  setup_all do
+    @lti_integration = create :lti_integration
+    fake_response_hash = {
+      access_token: 'fake_access_token',
+      exp: 1.hour.from_now.to_i,
+    }
+    @fake_access_token_response = OpenStruct.new({body: JSON.generate(fake_response_hash)})
+  end
+
+  test 'get_access_token calls the access_token_url from the LTI integration' do
+    HTTParty.stubs(:post).with(@lti_integration.access_token_url, anything, anything).returns(@fake_access_token_response)
+    jwt = get_access_token(@lti_integration.client_id, @lti_integration.issuer, ['test_scope', 'test_scope2'])
+    assert_equal 'fake_access_token', jwt
+  end
+
+  test 'get_access_token uses cached tokens if present' do
```

Comment:
We should probably also have a unit tests which exercises the expiration too. Look at other unit tests which freeze and move the clock forward a specific amount of time. I think I did some for CPA.